gen_cert.py

```diff
@@ -1274,6 +1125,22 @@ def _generate_v4_certificate(
             achievements_string = grade_interstitial.decode('utf-8').format(grade=grade_html) + '<br /><br />'
         achievements_paragraph = u"{0}{1}".format(achievements_string, achievements_description_string)
 
+        # Overide achievements/interstitial strings with yaml designations info based on designation
+        designation_tag = ''
+        if designation:
+            # Add student name designation if it exists
+            student_name = u"{name}, {designation}".format(
+                name=student_name,
+                designation=designation.decode('utf-8'),
+            )
+            if 'designations' in self.cert_data:
+                for key, value in self.cert_data.get('designations').iteritems():
```

Comment:
If we're explicitly checking for the presence of the key in the previous step, we shouldn't use the "safe" `dict.get` here; `dict.get` itself _also_ checks if the key exists, so this effectively compiles down to:

```python
def get(self, key, default=None):
    if key in self:
        return self[key]
    else:
        return default
```

so you end up with something like:

```python
if 'designations' in self.cert_data:
    if 'designations' in self.cert_data:
        for key, value in self.cert_data['designations'].iteritems():
            pass
```

We have two options here:
1. Keep the `in` check and index into the collection directly, or
2. Drop the `in` check and safely coalesce to an empty collection, when needed

# The Former
```python
if 'designations' in self.cert_data:
    for key, value in self.cert_data['designations'].iteritems():
        pass
```

or 

# The Latter
```python
# Note: we're passing in an empty dictionary `{}` as the default value
for key, value in self.cert_data.get('designations', {}).iteritems():
    pass
```

I can live with either approach, though I prefer the latter.
Not only is it less code, but it's also one less level of indenting.
All in all, a bit easier to read.